sdk/core/Azure.Core.Experimental/src/ISupportsProofOfPossession.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// An interface used to decorate a <see cref="TokenCredential"/> that supports <see href="https://learn.microsoft.com/entra/msal/dotnet/advanced/proof-of-possession-tokens">Proof of Possession tokens</see> for authenticating to Microsoft Entra ID.
+    /// </summary>
+    public interface ISupportsProofOfPossession
```

Comment:
Curious: we use an interface here because we want to add a method without introducing a base type?  

Given the challenges around evolving interfaces, did we consider other approaches to the design that might allow decoration without adding a type we can't really change in the future?  (I am ok if the answer is -- no other ones will work because of ___!)

Reply:
It's an interface because we need to decorate types that already have a base type (`TokenCredential`). Additionally, this is temporary to avoid changing `TokenRequestContext`.

Reply:
Makes sense!  Will it stay an interface when it moves to Azure.Core?

Reply:
(Also feel free to resolve this thread now!)

Reply:
Very cool.  Do you have a sketch of what the feature might look like when it moves into Core for GA, or is that still in the design stages?  (Just a curiosity question)

Reply:
`PopTokenRequestContext` will merge into `TokenRequestContext` and the credential will know what to do based on which properties are used on TRC when `GetToken` is called.